src/compile/invalid/datasources.ts

```diff
@@ -0,0 +1,63 @@
+import {MarkInvalidDataMode} from '../../invalid';
+
+import {DataSourceType} from '../../data';
+import {normalizeInvalidDataMode} from './normalizeInvalidDataMode';
+
+type PreOrPostFilteringInvalidValues = 'pre-filter' | 'post-filter';
```

Comment:
Are `include-null` and `exclude-null` easier to understand?

Reply:
will rename to 'include/exclude-invalid-values' -- it's not just Null, Infinity and NaNs are invalid too.